test/alternative_spec.rb

```diff
@@ -0,0 +1,51 @@
+require_relative "spec_helper.rb"
+
+describe "UpdateAlternatives::Alternative" do
+
+  context "when we get all alternatives names" do
+    it "should return an array with all alternatives names" do
+
+      alternatives_names_stub
+      expected_names = %w(pilconvert pip rake rake.ruby2.1 rdoc rdoc.ruby2.1 ri ri.ruby2.1)
+
+      alternatives_name_list = UpdateAlternatives::Alternative.all_names
+
+      expect(alternatives_name_list).to eq expected_names
+    end
+  end
+
+  describe ".all" do
+    subject { UpdateAlternatives::Alternative.all }
+
+    it "returns an array of Alternative objects" do
+      alternatives_pip_with_two_choices_stub
+      expect(subject.class).to eq Array
+      expect(subject.all? { |e| e.is_a?(UpdateAlternatives::Alternative) }).to eq true
+    end
+
+    context "alternative with one choice" do
+      it "is loaded correctly" do
```

Comment:
sorry, but this is very useless specification. IT should look like
```
context "alternative with one choice" do
  it "produce Alternative object with single choice in choices" do
```

same below. As I said, try to generate `rspec --format doc` and see if it is helpful specification.
To get example how it can look like is e.g. this one: 

```
jreidinger@pepa:~/yast/bootloader/test> rspec --format doc sections_test.rb 

Bootloader::Sections
  #all
    returns list of all available sections
  #default
    gets name of default section stored in grub2
    gets first section if nothing stored in grub2
    gets value written by #default=
  #default=
    sets new value for default
  #write
    writes default value
```

so see e.g. how default method specify how it can obtain its value.